src/core/Global.ttslua

```diff
@@ -283,6 +284,134 @@ function getTokenIndex(token)
   end
 end
 
+function activeRedrawEffect(originParams)
+  dataFromLastReplacementEffect = originParams
+end
+
+function makeButtonsToRedraw()
+  if isTokenXMLActive == true then
+    broadcastToAll("Clear already active buttons first, then try again", "Red")
+    return
+  end
+
+  local tokensInPlay = getChaosTokensinPlay()
+  if #tokensInPlay == 0 then
+    broadcastToAll("No tokens found in play area", "Red")
+    return
+  end
+
+  local chaosbag = findChaosBag()
+  local matchingTokensInPlay = {}
+  -- determine if only some tokens are able to be returned to the bag
+  if dataFromLastReplacementEffect.redrawEligibilty == "ineligible" then
```

Comment:
As talked about, I'd rather implement a system that supports both "valid" and "invalid" tokens similar to the "CardsThatSealTokens"